### PR TITLE
Label and translation adjustments in attributes.json

### PR DESCRIPTION
**Pull request by Shaky212**
The following major changes have been applied to attributes.json and it's scheme:

#1 Labels and info attributes must be different for oidcng entities and saml20 entities
#2 No distinction between the translations for the edit screen and info screen.

see: https://www.pivotaltracker.com/story/show/183736168

### CODE REVIEW

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Validator/Constraints/ValidAttribute.php [hunk 1]
```diff
@@ -31,11 +32,18 @@ class ValidAttribute extends Constraint
...
      * @return string
      */
     public function validatedBy()
     {
         return 'valid_attribute';
     }
+
+    public function getDefaultOption()
+    {
+        return 'type';
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Validator/Constraints/ValidAttribute.php:47]: I don't get this, does this indicate it should look to the `type` field for the default option value?

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Validator/Constraints/ValidAttribute.php [hunk 2]
```diff
@@ -31,11 +32,18 @@ class ValidAttribute extends Constraint
 
     public $messageAttributeMotivationNotSet = 'validator.attribute.motivation_not_set';
 
+    public $type = [Constants::TYPE_SAML, Constants::TYPE_OPENID_CONNECT_TNG];
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Validator/Constraints/ValidAttribute.php:35]: These seem to be the allowed values for the type field, is that correct? 

If that's the case. Would these values not be overwritten when setting the `type` on the command? I'd expect these values to be used to verify we are setting a valid value for `type`.

File: src/Surfnet/ServiceProviderDashboard/Application/Service/AttributeService.php [hunk 3]
```diff
@@ -90,31 +91,75 @@ public function getUrns(): array
...
+        Attribute $attribute,
+        string $entityType
+    ) {
+        switch ($entityType) {
+            case Constants::TYPE_SAML:
+                return $attribute->getSaml20Label();
+            case Constants::TYPE_OPENID_CONNECT_TNG:
+                return $attribute->getOidcngLabel();
+            default:
+                return '';
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Application/Service/AttributeService.php:134]: 1. Returning an empty string when the entity type is not supported feels like a situation to create strange situations. This look like a great situation to be throw an exception instead. (also goes for the other getters below)
2. Please add a return type definition on these private methods.

**Review by MKodde — CHANGES_REQUESTED** [review]

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Validator/Constraints/ValidAttribute.php (hunk 1 above)
Comment (Shaky212) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Validator/Constraints/ValidAttribute.php]: It is a dev left-over

**Review by MKodde — APPROVED** [review]
Looks good! See my one question. Maybe this is irrelevant. You decide.

File: app/Resources/schemas/attributes.scheme.json [hunk 4]
```diff
@@ -1,149 +1,101 @@
...
 					"en"
 				],
-				"patternProperties": {
-					"^en$": {
-						"$id": "#root/items/detail/en", 
-						"title": "En", 
+				"properties": {
+					"en": {
+						"$id": "#root/items/translations/en",
+						"title": "En",
```
[the remark this answers is not in the document]
Comment (Shaky212) [reply app/Resources/schemas/attributes.scheme.json]: New languages can be entered, but are not checked. Introduce patternProperties here will help to match other 2 character patterns to indicate a language and check the properties at a lower level.

**Review by MKodde — APPROVED** [review]
Nice work, lets go ahead with this.

Some CI observations: 

1. the security checkers should not be an allowed failure
2. both yarn audit and the composer audit tasks fail in this build. If this is fixed before a new 4.3 release, thats fine with me. But be aware of this!

**Pull request merged**